code/src/cljs/sixsq/nuvla/ui/edges/events.cljs

```diff
@@ -97,33 +107,113 @@
                                                            :event     [::get-nuvlabox-cluster
                                                                        (str "nuvlabox-cluster/" cluster-id)]}]]]}))
 
+(defn- get-full-filter-string
+  [{:keys [::spec/state-selector
+           ::spec/additional-filter] :as db}]
+  (general-utils/join-and
+   "id!=null"
+   (when state-selector (utils/state-filter state-selector))
+   additional-filter
+   (full-text-search-plugin/filter-text
+    db [::spec/edges-search])))
 
 (reg-event-fx
   ::get-nuvlaboxes
-  (fn [{{:keys [::spec/state-selector
-                ::spec/ordering
-                ::spec/additional-filter] :as db} :db} _]
+  (fn [{{:keys [::spec/ordering] :as db} :db} _]
     (let [ordering (or ordering spec/default-ordering)]
       {::cimi-api-fx/search
        [:nuvlabox
         (->> {:orderby (ordering->order-string ordering)
-              :filter  (general-utils/join-and
-                         (when state-selector (utils/state-filter state-selector))
-                         additional-filter
-                         (full-text-search-plugin/filter-text
-                           db [::spec/edges-search]))}
+              :filter   (get-full-filter-string db)}
              (pagination-plugin/first-last-params
-               db [::spec/pagination]))
+              db [::spec/pagination]))
         #(dispatch [::set-nuvlaboxes %])]})))
 
+(reg-event-fx
+ ::get-edges-without-edit-rights
+  (fn [{{:keys [::spec/select
+                ::session-spec/session] :as db} :db} _]
+    (let [selected-filter (table-plugin/build-bulk-filter
+                           select
+                           (get-full-filter-string db))
+          filter (general-utils/join-and
+                  (apply
+                   general-utils/join-and
+                   (map (fn [role]
+                          (str "acl/edit-meta!='" role "'"))
+                        (session-utils/get-roles session)))
+                  selected-filter)]
+      {::cimi-api-fx/search
+       [:nuvlabox
+        {:filter filter :select "id"}
+        #(dispatch [::set-edges-without-edit-rights %])]})))
+
+(reg-event-fx
+ ::set-edges-without-edit-rights
+ (fn [{:keys [db]} [_ nuvlaboxes]]
+   (if (instance? js/Error nuvlaboxes)
+     (dispatch [::messages-events/add
+                (let [{:keys [status message]} (response/parse-ex-info nuvlaboxes)]
+                  {:header  (cond-> (str "failure getting nuvlaboxes")
+                              status (str " (" status ")"))
+                   :content message
+                   :type    :error})])
+     {:db (assoc db ::spec/edges-without-edit-rights nuvlaboxes)})))
+
+
+(reg-event-fx
+ ::get-edges-tags
+ (fn [_ _]
+   {::cimi-api-fx/search
+    [:nuvlabox
+     {:select "tags"}
```

Comment:
This call will fetch tags of first 20 nuvlaboxes. Is that the purpose. There is an aggregation query that will get the 10 tags that are mostly used.

Reply:
I want all tags of edges that a user can see. And it returns tags for all edges, not just the first 20.

But just checked and the response for admins is 3MB, because per default metadata gets added to the resources.

So definitely better to use an aggregation here.

Can you point me to it? Haven't done any ES aggregations yet.

Reply:
This is an example query. The result is not visible in the UI. Check the response in browser network tab. 
https://nui.localhost/ui/api/nuvlabox?first=0&last=0&filter=&orderby=&select=&aggregation=terms:tags

example of response:
```{
  "count" : 24,
  "aggregations" : {
    "terms:tags" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [ {
        "key" : "nuvlabox=True",
        "doc_count" : 4
      }, {
        "key" : "a",
        "doc_count" : 1
      }, {
        "key" : "b",
        "doc_count" : 1
      }, {
        "key" : "blue",
        "doc_count" : 1
      }, {
        "key" : "new_tag",
        "doc_count" : 1
      } ]
    }
  },
  "acl" : {
    "query" : [ "group/nuvla-user" ],
    "add" : [ "group/nuvla-user" ],
    "bulk-action" : [ "group/nuvla-user" ]
  },
  "resource-type" : "nuvlabox-collection",
  "id" : "nuvlabox",
  "resources" : [ ],
  "operations" : [ {
    "rel" : "add",
    "href" : "nuvlabox"
  } ]
}
```